WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m

```diff
@@ -38,6 +44,17 @@ - (BOOL)fb_goToHomescreenWithError:(NSError **)error
 
 - (NSData *)fb_screenshot
 {
+  NSData *result = nil;
+  #if defined(__has_include)
+    #if __has_include("XCTest/XCUIScreen.h")
+  if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"11.0")) {
+    result = [[[XCUIScreen mainScreen] screenshot] PNGRepresentation];
```

Comment:
do you have an example? I'm not an expert in llvm preprocessing, but this was the single approach, that worked for both Xcode8 and Xcode9. The main problem there is that the whole _XCTest/XCUIScreen.h_ header file was not present prior to Xcode9. Also, if I try to call this method on iOS version < 11 with Xcode9 then it returns a valid PNG image, but the content of it is just an empty screen.